src/infer_trace.h

```diff
@@ -63,10 +63,12 @@ class InferenceTrace {
   const std::string& ModelName() const { return model_name_; }
   int64_t ModelVersion() const { return model_version_; }
   const std::string& RequestId() const { return request_id_; }
+  const std::string& Context() const { return context_; }
 
   void SetModelName(const std::string& n) { model_name_ = n; }
   void SetModelVersion(int64_t v) { model_version_ = v; }
   void SetRequestId(const std::string& request_id) { request_id_ = request_id; }
+  void SetContext(const std::string& context) { context_ = context; }
```

Comment:
I'm a little dubious of the user-provided `const char*` from `TRITONSERVER_InferenceTraceSetContext` being passed through to a `const std::string& context` here, but according to [this](https://stackoverflow.com/questions/67480842/does-stdstring-const-parameter-copies-the-passed-const-char) there will be a temp string constructed in between and then we're copying to the class member variable `context_` anyway so I think it's OK.

No action item here, just commenting a new thing I learned, and I was a little surprised the compiler is OK with this. I was surprised you can go from `const char*` to `const std::string&` because the `const char*`'s value can be changed, and then this "const ref string" would not be constant if referring to the original `const char*` provided.